### PR TITLE
minor bug fixes for multiple host zones

### PR DESCRIPTION
**Pull request by asalzmann**
## What is the purpose of the change

Fix a few small bugs to get multiple host zones working.

## Brief Changelog

- Update the REV_ACCT
- return nil (not error) if stSupply is 0
- use REDEMPTION var (instead of "redemption")

### CODE REVIEW

File: x/mint/types/params.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func NewParams(
 func DefaultParams() Params {
 	return Params{
 		MintDenom:               sdk.DefaultBondDenom,
-		GenesisEpochProvisions:  sdk.NewDec(6_700_000).QuoInt64(365),
+		GenesisEpochProvisions:  sdk.NewDec(6_700_000_000_000),
```
Comment (asalzmann) [inline x/mint/types/params.go:53]: @riley-stride did you make this change?

File: x/stakeibc/keeper/hooks.go [hunk 2]
```diff
@@ -259,13 +259,12 @@ func (k Keeper) StakeExistingDepositsOnHostZones(ctx sdk.Context, epochNumber ui
 			processAmount := utils.Int64ToCoinString(depositRecord.Amount, hostZone.HostDenom)
 			amt, err := sdk.ParseCoinNormalized(processAmount)
 			if err != nil {
-				k.Logger(ctx).Error(fmt.Sprintf("Could not process coin %s: %s", hostZone.HostDenom, err))
+				k.Logger(ctx).Error(fmt.Sprintf("Could not process coin %s: %s", hostZone.HostDenom, err.Error()))
```
Comment (vish-stride) [inline x/stakeibc/keeper/hooks.go:262]: extreme paranoia but should we make this %v

File: x/mint/types/params.go (hunk 1 above)
Comment (asalzmann) [reply x/mint/types/params.go:53]: iirc you mentioned this param yesterday, not sure what it does

File: x/stakeibc/keeper/callbacks.go [hunk 3]
```diff
@@ -157,18 +157,11 @@ func WithdrawalBalanceCallback(k Keeper, ctx sdk.Context, args []byte, query icq
...
-	}
-	if reinvestCoin.Amount.Int64() > 0 {
-		msgs = append(msgs, &banktypes.MsgSend{FromAddress: withdrawalAccount.GetAddress(),
-			ToAddress: delegationAccount.GetAddress(), Amount: sdk.NewCoins(reinvestCoin)})
-	}
-	// if both amounts are 0, return an error
-	if strideCoin.Amount.Int64() <= 0 && reinvestCoin.Amount.Int64() <= 0 {
-		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "No amount to send in withdrawal callback.")
-	}
+	msgs = append(msgs, &banktypes.MsgSend{FromAddress: withdrawalAccount.GetAddress(),
```
Comment (vish-stride) [inline x/stakeibc/keeper/callbacks.go:160]: do we no longer need this check?

File: x/stakeibc/keeper/callbacks.go [hunk 4]
```diff
@@ -157,18 +157,11 @@ func WithdrawalBalanceCallback(k Keeper, ctx sdk.Context, args []byte, query icq
 
 	var msgs []sdk.Msg
 	// construct the msg
-	if strideCoin.Amount.Int64() > 0 {
```
Comment (asalzmann) [inline x/stakeibc/keeper/callbacks.go:160]: Think we actually do want the first two checks here

**Review by vish-stride — APPROVED** [review]
one minor question, otherwise looks good!

File: x/mint/types/params.go (hunk 1 above)
Comment (riley-stride) [inline x/mint/types/params.go:53]: let's revert this, I think I made this change by mistake while trying to make a change to osmosis' mint params
```suggestion
		GenesisEpochProvisions:  sdk.NewDec(6_700_000).QuoInt64(365),
```

**Review by riley-stride — CHANGES_REQUESTED** [review]
one change request

File: x/stakeibc/keeper/callbacks.go (hunk 3 above)
Comment (asalzmann) [reply x/stakeibc/keeper/callbacks.go]: @riley-stride is adding the checks back

File: x/stakeibc/keeper/callbacks.go [hunk 5]
```diff
@@ -127,7 +127,7 @@ func WithdrawalBalanceCallback(k Keeper, ctx sdk.Context, args []byte, query icq
 	withdrawalAccount := zone.GetWithdrawalAccount()
 	delegationAccount := zone.GetDelegationAccount()
 	// TODO(TEST-112) set the stride revenue address in a config file
-	REV_ACCT := "cosmos1wdplq6qjh2xruc7qqagma9ya665q6qhcwju3ng"
+	REV_ACCT := zone.GetFeeAccount().GetAddress()
```
Comment (sampocs) [inline x/stakeibc/keeper/callbacks.go:130]: nit: revenueAccount

File: x/mint/types/params.go (hunk 1 above)
Comment (riley-stride) [reply x/mint/types/params.go]: addressed this

**Review by riley-stride — APPROVED** [review]

File: x/mint/types/params.go (hunk 1 above)
Comment (asalzmann) [reply x/mint/types/params.go:53]: done

File: x/stakeibc/keeper/callbacks.go [hunk 6]
```diff
@@ -125,9 +125,20 @@ func WithdrawalBalanceCallback(k Keeper, ctx sdk.Context, args []byte, query icq
 	k.Logger(ctx).Info(fmt.Sprintf("ICA Bank Sending %d%s from withdrawalAddr to delegationAddr.", coin.Amount.Int64(), coin.Denom))
 
 	withdrawalAccount := zone.GetWithdrawalAccount()
+	if withdrawalAccount == nil {
+		k.Logger(ctx).Error(fmt.Sprintf("WithdrawalBalanceCallback: no withdrawal account found for zone: %s", zone.ChainId))
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "WithdrawalBalanceCallback: no withdrawal account found for zone: %s", zone.ChainId)
```
Comment (vish-stride) [inline x/stakeibc/keeper/callbacks.go:130]: fmt.Sprintf

File: x/stakeibc/keeper/callbacks.go [hunk 7]
```diff
@@ -125,9 +125,20 @@ func WithdrawalBalanceCallback(k Keeper, ctx sdk.Context, args []byte, query icq
...
+	}
 	delegationAccount := zone.GetDelegationAccount()
-	// TODO(TEST-112) set the stride revenue address in a config file
-	REV_ACCT := "cosmos1wdplq6qjh2xruc7qqagma9ya665q6qhcwju3ng"
+	if delegationAccount == nil {
+		k.Logger(ctx).Error(fmt.Sprintf("WithdrawalBalanceCallback: no delegation account found for zone: %s", zone.ChainId))
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "WithdrawalBalanceCallback: no delegation account found for zone: %s", zone.ChainId)
```
Comment (vish-stride) [inline x/stakeibc/keeper/callbacks.go:135]: fmt.Sprintf

File: x/stakeibc/keeper/callbacks.go [hunk 8]
```diff
@@ -125,9 +125,20 @@ func WithdrawalBalanceCallback(k Keeper, ctx sdk.Context, args []byte, query icq
...
+	}
+	feeAccount := zone.GetFeeAccount()
+	if feeAccount == nil {
+		k.Logger(ctx).Error(fmt.Sprintf("WithdrawalBalanceCallback: no fee account found for zone: %s", zone.ChainId))
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "WithdrawalBalanceCallback: no fee account found for zone: %s", zone.ChainId)
```
Comment (vish-stride) [inline x/stakeibc/keeper/callbacks.go:140]: fmt.Sprintf

**Review by vish-stride — APPROVED** [review]
extremely nit, but we should wrap these in sprintf imo

File: x/stakeibc/keeper/callbacks.go (hunk 8 above)
Comment (riley-stride) [reply x/stakeibc/keeper/callbacks.go]: done

File: x/stakeibc/keeper/callbacks.go (hunk 7 above)
Comment (riley-stride) [reply x/stakeibc/keeper/callbacks.go]: done

File: x/stakeibc/keeper/callbacks.go (hunk 6 above)
Comment (riley-stride) [reply x/stakeibc/keeper/callbacks.go]: done

File: x/stakeibc/keeper/callbacks.go [hunk 9]
```diff
@@ -133,7 +145,7 @@ func WithdrawalBalanceCallback(k Keeper, ctx sdk.Context, args []byte, query icq
 	strideCommission := sdk.NewDec(stCommission).Quo(sdk.NewDec(100))
 	// check that stride commission is between 0 and 1
 	if strideCommission.LT(sdk.ZeroDec()) || strideCommission.GT(sdk.OneDec()) {
-		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "Aborting reinvestment callback -- Stride commission must be between 0 and 1!")
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Aborting reinvestment callback -- Stride commission must be between 0 and 1!"))
```
Comment (asalzmann) [inline x/stakeibc/keeper/callbacks.go:148]: nit: don't need sprintf given there's no variable replacement

**Review by asalzmann — COMMENTED** [review]
nice, much safer with these checks!

File: x/stakeibc/keeper/callbacks.go (hunk 9 above)
Comment (asalzmann) [reply x/stakeibc/keeper/callbacks.go:148]: linter got you too haha

Comment (asalzmann) [reply x/stakeibc/keeper/callbacks.go]: fixed

**Pull request merged**